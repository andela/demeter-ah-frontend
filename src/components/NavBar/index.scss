nav{
  img#logo {
    content:'';
    display:inline-block;
    padding: 1.2rem 1.2rem;
    width: 0;
    height: 0;
    background-repeat: no-repeat;
    background-size: contain;
    background-image: url("../../assets/images/brand-logo.png") ; /* lets say 20x20 */
    @media (min-width: 768px){
      padding: 1rem 6rem;
      background-image: url("../../assets/images/logo.png") ;
    }
  }
  button {
    padding: .6rem 1rem !important;
  }
}