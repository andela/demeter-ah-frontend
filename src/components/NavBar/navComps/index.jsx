import React from 'react';
import { Link } from 'react-router-dom';
import Button from '../../Button';

export const AuthNav = () => (
  <>
    <Link to="/signup">
      <Button
        type="button"
        name="Sign Up"
        classes="btn-purple w-32 hidden md:inline"
      />
    </Link>
    <Link to="/signin">
      <Button
        type="button"
        name="Sign In"
        classes=" cursor-pointer btn-white w-32 ml-2 text-purple-650 border hidden md:inline border-solid border-purple-650"
      />
    </Link>
  </>
);

export const CreateArticleNav = ({ openModal }) => (
  <Button type="button" onClick={() => openModal()} name="Next Step" classes="btn-purple" />
);
