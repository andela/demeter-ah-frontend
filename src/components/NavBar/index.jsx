import React, { useState, Fragment } from 'react';
import { connect } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';
import * as actions from '../../store/actions/articles';
import { signOutAction } from '../../store/actions/signOut';
import { AuthNav, CreateArticleNav } from './navComps';
import logo from '../../assets/images/logo.png';
import UserNavInfo from '../UserNavInfo';
import Search from '../Search';
import Dropdown from '../Dropdown';

const NavBar = ({
  history, isAuthenticated, openModal, onSignOut
}) => {
  const [toggle, setToggle] = useState(false);
  const path = history.location.pathname;

  const handleDropDown = () => {
    setToggle(prevState => !prevState);
  };

  const handleSignOut = () => {
    onSignOut();
    return (<Redirect to="/" />);
  };

  const showNavComp = () => (
    <Fragment>
      <UserNavInfo onClick={handleDropDown} />
      {path === '/article/create' ? <CreateArticleNav history={history} openModal={openModal} /> : ''}
      {toggle ? <Dropdown handleDropDown={handleDropDown} handleSignOut={handleSignOut} /> : ''}
    </Fragment>
  );

  return (
    <nav className="flex items-center justify-between relative flex-wrap bg-white-500 pt-2 p-3 z-40 shadow-md">
      <div className="flex items-center flex-shrink-0 text-dark mr-6">
        <Link to="/">
          <img id="logo" className="w-32" src={logo} alt="Author's Haven" />
        </Link>
      </div>
      <div className="flex relative">
        <Search />
        {isAuthenticated ? showNavComp() : <AuthNav />}
      </div>
    </nav>
  );
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
});

export default connect(mapStateToProps, {
  openModal: actions.openPublishModal,
  onSignOut: signOutAction,
})(NavBar);
