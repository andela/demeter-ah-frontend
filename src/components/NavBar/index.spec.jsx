import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from './index';

describe('NavBar Component', () => {
  it('Should render without errors', () => {
    const component = shallow(<NavBar />);
    expect(component).toMatchSnapshot();
  });

  it('Should render along with children componenet', () => {
    const component = mount(
      <Router>
        <NavBar />
      </Router>
    );
    expect(component).toMatchSnapshot();
    expect(component.find('button')).toHaveLength(2);
    expect(component.find('img')).toHaveLength(1);
  });
});
