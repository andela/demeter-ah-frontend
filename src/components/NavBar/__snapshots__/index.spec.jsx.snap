// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component Should render along with children component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(NavBar)
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <NavBar
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isAuthenticated={false}
          onSignOut={[Function]}
          openModal={[Function]}
        >
          <nav
            className="flex items-center justify-between relative flex-wrap bg-white-500 pt-2 p-3 z-40 shadow-md"
          >
            <div
              className="flex items-center flex-shrink-0 text-dark mr-6"
            >
              <Link
                to="/"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  <img
                    alt="Author's Haven"
                    className="w-32"
                    id="logo"
                    src="test-file-stub"
                  />
                </a>
              </Link>
            </div>
            <div
              className="flex relative"
            >
              <search>
                <div
                  className="search relative flex items-center justify-center p-0 mr-6"
                >
                  <input
                    className="border hidden md:inline-flex border-solid border-gray-350 bg-gray-20 rounded-4xl h-9 w-64 pr-2 pl-4"
                    placeholder="Search..."
                    type="text"
                  />
                  <searchIcon
                    classes="searchIcon cursor-pointer absolute right-0"
                  >
                    <svg
                      className="searchIcon cursor-pointer absolute right-0"
                      fill="none"
                      height="20"
                      viewBox="0 0 112 113"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M109.339 102.925L81.9692 74.9538C89.0066 66.7336 92.8624 56.3909 92.8624 45.6237C92.8624 20.4672 72.033 0 46.4312 0C20.8294 0 0 20.4672 0 45.6237C0 70.7802 20.8294 91.2474 46.4312 91.2474C56.0425 91.2474 65.2015 88.3989 73.0322 82.9915L100.61 111.175C101.763 112.351 103.313 113 104.975 113C106.548 113 108.039 112.411 109.172 111.34C111.578 109.064 111.655 105.292 109.339 102.925ZM46.4312 11.9018C65.3549 11.9018 80.7499 27.0291 80.7499 45.6237C80.7499 64.2183 65.3549 79.3456 46.4312 79.3456C27.5075 79.3456 12.1125 64.2183 12.1125 45.6237C12.1125 27.0291 27.5075 11.9018 46.4312 11.9018Z"
                        fill="#606060"
                      />
                    </svg>
                  </searchIcon>
                </div>
              </search>
              <AuthNav>
                <Link
                  to="/signup"
                >
                  <a
                    href="/signup"
                    onClick={[Function]}
                  >
                    <Button
                      classes="btn-purple w-32 hidden md:inline"
                      name="Sign Up"
                      type="button"
                    >
                      <button
                        className="btn btn-purple w-32 hidden md:inline"
                        disabled={false}
                        type="button"
                      >
                        Sign Up
                      </button>
                    </Button>
                  </a>
                </Link>
                <Link
                  to="/signin"
                >
                  <a
                    href="/signin"
                    onClick={[Function]}
                  >
                    <Button
                      classes=" cursor-pointer btn-white w-32 ml-2 text-purple-650 border hidden md:inline border-solid border-purple-650"
                      name="Sign In"
                      type="button"
                    >
                      <button
                        className="btn  cursor-pointer btn-white w-32 ml-2 text-purple-650 border hidden md:inline border-solid border-purple-650"
                        disabled={false}
                        type="button"
                      >
                        Sign In
                      </button>
                    </Button>
                  </a>
                </Link>
              </AuthNav>
            </div>
          </nav>
        </NavBar>
      </Connect(NavBar)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`NavBar Component Should render without errors 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(NavBar)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</Router>
`;
